app/models/project.rb

```diff
@@ -1,4 +1,19 @@
 class Project < ApplicationRecord
   validates_presence_of :name, :material
+
   belongs_to :challenge
+  has_many :contestant_projects
+  has_many :contestants, through: :contestant_projects
+
+  def contestant_count
+    self.contestants.count
+  end
+
+  def contestant_experience_avg
+    if self.contestant_count == 0
+      0
+    else
+      self.contestants.average(:years_of_experience)
+    end
+  end
```

Comment:
Nice job with these methods, especially taking into account having no contestants. I think you could also remove the `self.` and they should still work.